site/content/integrate/incoming-webhooks/_index.md

```diff
@@ -0,0 +1,92 @@
+---
+title: "Incoming Webhooks"
+date: "2017-08-19T12:01:23-04:00"
+section: "integrate"
+---
+
+# Incoming Webhooks
+
+Incoming webhooks let you POST some data to a Mattermost endpoint to create a message in a channel.
+
+### Basic Usage
+
+Follow [the admin guide](https://docs.mattermost.com/developer/webhooks-incoming.html#simple-incoming-webhook) to create the webhook endpoint. It'll look something like this:
+
+```
+https://your-mattermost-server.com/hooks/xxx-generatedkey-xxx
+```
+
+__Treat this endpoint as a secret.__ Anyone who has it will be able to post messages to your Mattermost instance.
+
+To use the endpoint, have your application make the following request:
+
+```http
+POST /hooks/xxx-generatedkey-xxx HTTP/1.1
+Host: your-mattermost-server
+Content-Type: application/json
+Content-Length: 63
+
+{"text": "Hello, this is some text\nThis is more text. :tada:"}
+```
+
+As a curl:
+
+```bash
+curl -i -X POST -H 'Content-Type: application/json' -d '{"text": "Hello, this is some text\nThis is more text. :tada:"}' http://{your-mattermost-site}/hooks/xxx-generatedkey-xxx
+```
+
+For compatibility with Slack incoming webhooks, if no `Content-Type` header is set then the request body must be prefixed with `payload=`, like so:
+
+```bash
+payload={"text": "Hello, this is some text\nThis is more text. :tada:"}
+```
+
+A successful request will get the following response:
+
+```http
+HTTP/1.1 200 OK
+Content-Type: text/plain
+X-Request-Id: hoan6o9ws7rp5xj7wu9rmysrte
+X-Version-Id: 4.7.1.dev.12799cd77e172e8a2eba0f9091ec1471.false
+Date: Sun, 04 Mar 2018 17:19:09 GMT
+Content-Length: 2
+
+ok
+```
+
+All webhook posts will display a `BOT` indicator next to the username in Mattermost clients, to help prevent against [phishing attacks](https://en.wikipedia.org/wiki/Phishing).
+
+### Parameters
+
+Incoming webhooks support more than just the `text` field. Here is a full list of supported parameters.
+
+| Parameter | Description | Required |
+|---|---|---|
+| text | [Markdown-formatted](https://docs.mattermost.com/help/messaging/formatting-text.html) message to display in the post | If `attachments` is not set, yes |
+| channel | Overrides the channel the message posts in.<br> Use an "@" followed by a username to send to a direct message.<br> Defaults to the channel set during webhook creation. | No |
```

Comment:
As a developer, I'd want to know: can I post to any private channel if I know it by name?

Reply:
Yeah, good point. I'll add a note for that